src/KReduce.chpl

```diff
@@ -0,0 +1,110 @@
+/*
+ * 'kreduce' reduction implementation. Returns 
+ * array of the k extreme values of an array 
+ * of type eltType. Whether it will be the max
+ * or min values is specified through the "isMin"
+ * field (true gives min values, false gives max
+ * values).
+ *
+ * **NOTE**
+ * This reduction performs well with a small `k`
+ * value, but sees a significant drop off in
+ * performance as `k` grows larger, although the
+ * exact value for this threshold is system dependent.
+ * This is because this is a per-task reduciton, so
+ * as `k` increases, the amount of random access that
+ * occurs will increase, slowing down access. This
+ * operation is equivalent to performing an `argsort()`
+ * on an array, and then using a slice of these values
+ * to retrieve the first or last `k` indices from that
+ * sorted indice array.
+ */
+
+module KReduce {
+  use KExtreme;
+
+  class kreduce : ReduceScanOp {
```

Comment:
doc comments

Reply:
also a couple of comments on how this works with the chapel reduction framework.

Reply:
or a pointer to the chapel documentation which describes it.